src/os/source/vectorsource.js

```diff
@@ -534,6 +540,34 @@ os.source.Vector.prototype.changed = function() {
 };
 
 
+/**
+ * Returns the array of Mapping Managers
+ * @return {*}
+ */
+os.source.Vector.prototype.getMappingManagers = function() {
+  return this.mappingManagers;
+};
+
+
+/**
+ * Returns the mapping manager from a key
+ * @param {string} key
+ * @return {*}
+ */
+os.source.Vector.prototype.getMappingManager = function(key) {
+  return this.mappingManagers[key] || undefined;
+};
+
+
+/**
+ * @param {string} key
+ * @param {*} manager
+ */
+os.source.Vector.prototype.setMappingManager = function(key, manager) {
+  this.mappingManagers[key] = manager;
+};
+
```

Comment:
How much of this lives between refreshes? Should this.mappingManagers has some/part of itself saved during this.persist() and this.restore()?

Reply:
I think we'd just need to persist `this.MappingOptions_` and the source they're associated with. This would make it easy to recreate the whole mappingManager.

Reply:
Ideally the manager would implement `IPersistable` and take care of its own persistence internally.